### PR TITLE
Adding tensor names for TFLite Java

### LINKED ISSUE #27360: TFLite Named Inputs and Outputs
Using models with multiple inputs and outputs it would be helpful to have the names of the input and output channels available inside the Java and C++ APIs for TFLite.

**System information**
- TensorFlow version (you are using): latest
- Are you willing to contribute it (Yes/No): Yes

**Describe the feature and the current behavior/state.**

The model inputs and outputs can be listed and queried by size but not by name (no argument to get name of a `Tensor` or get input/output tensor names from an `Interpreter`. It would be very helpful for using models if you could get the names of the inputs and outputs particularly since many models have multiple of each (that are often not discriminated by size).

**Will this change the current api? How?**

It would just at methods `.getInputNames` and `.getOutputNames` to the `org.tensorflow.lite.Interpreter` class or add `.name` to the `org.tensorflow.lite.Tensor` class 

**Who will benefit with this feature?**

It would make it a lot easier to have backward compatible models (newer versions predict more features and code doesn't always have to use all of the outputs a model provides). It would also allow reusing the same model for different applications and have the code select the output with the correct name rather than hard coding an output index to take or checking by output size (my current solutions).

**Any Other info.**

### PR DESCRIPTION
**Pull request by kmader**
Add the commands to get the input and output names for TFLite java
- Closes #27360

### CODE REVIEW

File: tensorflow/lite/java/src/main/java/org/tensorflow/lite/NativeInterpreterWrapper.java [hunk 1]
```diff
@@ -268,6 +272,11 @@ int getInputTensorCount() {
     return inputTensors.length;
   }
 
+  /** Gets the names of the input tensors. */
+  String[] getInputTensorNames() {
```
Comment (lu-wang-g) [inline tensorflow/lite/java/src/main/java/org/tensorflow/lite/NativeInterpreterWrapper.java]: Better to check if the name exists.

Comment (kmader) [reply tensorflow/lite/java/src/main/java/org/tensorflow/lite/NativeInterpreterWrapper.java]: Sure, I think the `init` method has to be called, but probably not a bad idea to add a check anyways.

Comment (kmader) [reply tensorflow/lite/java/src/main/java/org/tensorflow/lite/NativeInterpreterWrapper.java]: Should I just add this block before?
```
    if (inputTensorNames == null) {
      return new String[]
    }
```

Comment (lu-wang-g) [reply tensorflow/lite/java/src/main/java/org/tensorflow/lite/NativeInterpreterWrapper.java]: Either way looks good to me. inputTensorNames=new String[] {} seems better. What do you think?

Comment (kmader) [reply tensorflow/lite/java/src/main/java/org/tensorflow/lite/NativeInterpreterWrapper.java]: Done!

Comment (kmader) [reply tensorflow/lite/java/src/main/java/org/tensorflow/lite/NativeInterpreterWrapper.java]: @lu-wang-g done!

**Review by lu-wang-g — APPROVED** [review]

**Comment by jdduke** [comment]
We should probably just add the name as a field on the Tensor class in Tensor.java, letting you do something like:
```
getInputTensor(0).name()
```

**Comment by jdduke** [comment]
The merge was a bit premature, expect a rollback shortly.

**Comment by kmader** [comment]
@jdduke that makes sense, but I saw there was more infrastructure already in place for `getInputTensorNames` (which the python tflite interpreter also has), so maybe it should have the names in the `Tensor` and `Interpreter`.

**Comment by jdduke** [comment]
Yeah, we're trying to streamline the Interpreter interface, particularly for properties that are Tensor-specific.

**Pull request merged**